genienlp/server.py

```diff
@@ -45,89 +45,125 @@
 from .util import set_seed, init_devices, load_config_json, log_model_size
 from .validate import generate_with_model
 
+from bootleg.annotator import Annotator
+from .data_utils.bootleg import Bootleg
+
 logger = logging.getLogger(__name__)
 
 
 class Server:
-    def __init__(self, args, numericalizer, model, device, confidence_estimator):
+    def __init__(self, args, numericalizer, model, device, confidence_estimator, bootleg_annotator=None):
         self.args = args
         self.device = device
         self.numericalizer = numericalizer
         self.model = model
         self.confidence_estimator = confidence_estimator
+        self.bootleg_annotator = bootleg_annotator
 
-        self._cached_tasks = dict()
+        self._cached_task_names = dict()
 
     def numericalize_examples(self, ex):
-        all_features = NumericalizedExamples.from_examples(ex, self.numericalizer)
+
+        all_features = NumericalizedExamples.from_examples(ex, self.numericalizer, self.args.add_types_to_text)
         # make a single batch with all examples
         return NumericalizedExamples.collate_batches(all_features, self.numericalizer, device=self.device)
+    
+    def bootleg_process_examples(self, ex, label, task):
+        line = {}
+        if task.is_contextual():
+            line['sentence'] = ex.question
+        else:
+            line['sentence'] = ex.context
```

Comment:
Wat?
If you need the task to tell you which field is the sentence, you ask the task to tell you. You don't hide this logic in a random place.
Please refactor.